zebra/zserv.c

```diff
@@ -568,7 +568,7 @@ static void zserv_client_free(struct zserv *client)
 
 		close(client->sock);
 
-		if (!client->gr_instance_count) {
+		if (DYNAMIC_CLIENT_GR_DISABLED(client)) {
```

Comment:
So, not immediately relevant to this pr - but should the GR de-init handling go in `zserv_close_client`? Conceptually, that is the handler for when a client disconnects, and `zserv_client_free` simply `free()`'s all the resources associated with it. Obviously the same argument can, and should, be made for the socket deinit in the same function. It's even weird that we have `hook_call(zserv_client_close, client)` in the free function and not in the function named `zserv_close_client`. Thoughts?

Reply:
> So, not immediately relevant to this pr - but should the GR de-init handling go in `zserv_close_client`? Conceptually, that is the handler for when a client disconnects, and `zserv_client_free` simply `free()`'s all the resources associated with it. Obviously the same argument can, and should, be made for the socket deinit in the same function. It's even weird that we have `hook_call(zserv_client_close, client)` in the free function and not in the function named `zserv_close_client`. Thoughts?

 Earlier thought of using zserv_client_free but then good amount of code was getting replicated or had to break the function to make use of common function, so went ahead and used zserv_client_free. To make it more readable I think your suggestion make sense. I will clean this part of code and raise a new PR.